src/etc/inc/filter.lib.inc

```diff
@@ -90,13 +90,23 @@ function filter_core_get_antilockout()
 {
     global $config;
 
+    // Migration from old noantilockout option
```

Comment:
Assumes the default (new) configuration contains a fixed choice, which circles back to a static `lan` in `config.xml.sample`,  when not persisting lan, initial boot after install will likely always persist `lan` here anyway, which might then be altered by an interface mismatch later on. 

When removing the side affect `write_config();`, a migration also will assume a static situation, which past experiences learn has its limitations.

It's also good to remember, you're only able to cause your issue by removing existing interfaces, which almost sound like if we didn't let you remove "WAN", you didn't have an issue either..... (probably not true, but similar to the changes proposed)

Just my two cents on the subject.

Reply:
Yes, that's what I [wrote in the related issue of this PR](https://github.com/opnsense/core/issues/7372#issuecomment-2054016915): The place and the way of the migration doesn't feel right, which is due to my lack of knowledge about the internal workings of the config system. A one-time migration is most likely the better choice, but I'd need guidance as how to do that.

And it is not the removal of the WAN interface that causes trouble, it is the removal of the LAN interface (more precisely: the interface that in the config is named "lan"). Once that is gone, the opt with the lowest number is chosen:

If that opt happens to act as a WAN => antilockout rule opens ports on that WAN
If that opt happens to be disabled => antilockout rule doesn't work at all